engine/pkg/externalresource/model/model.go

```diff
@@ -133,6 +137,12 @@ const (
 	ResourceTypeS3        = ResourceType("s3")
 )
 
+// BuildPrefix returns the prefix of the resource type.
+func (r ResourceType) BuildPrefix() string {
+	// For local file, the prefix is `/local%`. For S3, the prefix is `/s3%`.
+	return fmt.Sprintf("/%s%%", r)
```

Comment:
If I understand correctly, the suffix `%` is used in SQL queries. It's not a good idea to coupling with SQL here.